sdk/src/wallet/core/operations/stronghold_backup/mod.rs

```
@@ -255,13 +233,16 @@ impl Wallet<StrongholdSecretManager> {
             return Err(crate::wallet::Error::Backup("backup path doesn't exist"));
         }
 
-        let mut accounts = self.accounts.write().await;
+        // Will be replaced by the restored wallet data
+        let mut wallet_data = self.data_mut().await;
+
+        // TODO: Is there a way to ensure that the user can't mess up?
```

Maybe return an error if unspent outputs aren't empty